core/src/main/java/com/yahoo/tagchowder/ParserContext.java

```diff
@@ -0,0 +1,85 @@
+/*
+ *
+ * ====================================================================
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  ====================================================================
+ */
+
+/*
+ * Changes to the original project are Copyright 2019 Oath Inc.
+ */
+
+package com.yahoo.tagchowder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class to store the current context/state of the Parser object.
+ * @author manishsingh
+ */
+public final class ParserContext {
+    /**
+     * Flag to decide whether to do string intern or not.
+     */
+     private boolean useIntern = true;
+    /**
+     * Hash Map to be used when not using string intern.
+     */
+    private Map<String, Boolean> stringPoolMap  = new HashMap<>();
+
+     /**
+      * Constructor.
+      * @param useIntern whether to use string intern or not
+      */
+    public ParserContext(final boolean useIntern) {
+          this.useIntern = useIntern;
+     }
+
+    /**
+     * Get the useIntern flag value.
+     * @return useIntern flag value
+     */
+     public boolean getUseIntern() {
+          return useIntern;
+     }
+
+    /**
+     * Set the use intern flag.
+     * @param useIntern whether to use string intern or not
+     */
+     public void setUseIntern(final boolean useIntern) {
```

Comment:
final

Reply:
class itself is final. I get checkstyle errors if I add final to methods.